src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionMidEventHandler.java

```diff
@@ -138,11 +147,98 @@ protected DirectionOrDecisionIssuedTemplateBody createPayload(SscsCaseData caseD
             builder.dateOfDecision(caseData.getWriteFinalDecisionDateOfDecision());
         }
 
+        builder.isIndefinite(caseData.getWriteFinalDecisionEndDate() == null);
+        builder.endDate(caseData.getWriteFinalDecisionEndDate());
+        builder.startDate(caseData.getWriteFinalDecisionStartDate());
+        builder.appellantName(buildName(caseData));
+
+        setEntitlements(builder, caseData);
+
+        setDescriptorsAndPoints(builder, caseData);
+
         DirectionOrDecisionIssuedTemplateBody payload = builder.build();
         validateRequiredProperties(payload);
+
         return payload;
     }
 
+    private void setEntitlements(DirectionOrDecisionIssuedTemplateBodyBuilder builder, SscsCaseData caseData) {
+
+        String dailyLivingAwardType = caseData.getPipWriteFinalDecisionDailyLivingQuestion();
+        String mobilityAwardType = caseData.getPipWriteFinalDecisionMobilityQuestion();
+
+        if (AwardType.ENHANCED_RATE.getKey().equals(dailyLivingAwardType)) {
+            builder.dailyLivingIsEntited(true);
+            builder.dailyLivingAwardRate("enhanced rate");
```

Comment:
Minor thing but you could just do this in one place rather than set the dailyAwardRate in each if else block `StringUtils.splitByCharacterTypeCamelCase(dailyLivingAwardType),`
                ' '
        ).toLowerCase() `

Reply:
Thanks Jack - I've now replaced the two calls as suggested